tcl/filter.tcl

```diff
@@ -84,7 +87,9 @@ proc qc::filter_authenticate {event {error_handler qc::error_handler}} {
             
             # Roll the anonymous session after 1 hour.
             if {[qc::auth] == [qc::anonymous_user_id]} {
-                qc::cookie_set session_id [qc::anonymous_session_id]
+                global session_id
+                set session_id [qc::anonymous_session_id]
+                qc::cookie_set session_id $session_id
             }
```

Comment:
Not sure this will work because cookie_set only sets outbound reply http headers while cookie_get reads received headers.